### PR TITLE
arpack: 3.7.0 -> 3.8.0

### PR DESCRIPTION
**Pull request by dotlambda**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/science/math/arpack/default.nix [hunk 1]
```diff
@@ -1,17 +1,25 @@
...
   src = fetchFromGitHub {
     owner = "opencollab";
     repo = "arpack-ng";
     rev = version;
-    sha256 = "1x7a1dj3dg43nlpvjlh8jzzbadjyr3mbias6f0256qkmgdyk4izr";
+    sha256 = "sha256-nc710iLRqy/p3EaVgbEoCRzNJ9GpKqqQp33tbn7R6lA=";
   };
 
+  patches = [
+    (fetchpatch {
```
Comment (doronbehar) [inline pkgs/development/libraries/science/math/arpack/default.nix:16]:
```suggestion
    # https://github.com/opencollab/arpack-ng/pull/301
    (fetchpatch {
```

**Review by doronbehar — CHANGES_REQUESTED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 117685` at 3bb50d3f run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>julia_1</li>
    <li>julia_10</li>
    <li>octave-jit</li>
    <li>openems</li>
    <li>python38Packages.python-csxcad</li>
    <li>python38Packages.python-openems</li>
    <li>python39Packages.python-csxcad</li>
    <li>python39Packages.python-openems</li>
  </ul>
</details>
<details>
  <summary>6 packages skipped due to time constraints:</summary>
  <ul>
    <li>gama</li>
    <li>julia (julia_15)</li>
    <li>librsb</li>
    <li>minc_widgets</li>
    <li>octave</li>
    <li>octavePackages.audio</li>
  </ul>
</details>
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>arpack</li>
    <li>calculix</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/development/libraries/science/math/arpack/default.nix:16:5:
```
   |
16 |     (fetchpatch {
   |     ^
```
</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 117685` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
6 packages marked as broken and skipped:
</summary>
<ul>
<li>julia_1</li>
<li>julia_10</li>
<li>octave-jit</li>
<li>octavePackages.fem-fenics</li>
<li>octavePackages.vibes</li>
<li>octavePackages.vrml</li>
</ul>
</details>
<details>
<summary>
79 packages built:
</summary>
<ul>
<li>arpack</li>
<li>calculix</li>
<li>gama</li>
<li>julia (julia_15)</li>
<li>julia_13</li>
<li>librsb</li>
<li>minc_widgets</li>
<li>octave</li>
<li>octaveFull</li>
<li>octavePackages.arduino</li>
<li>octavePackages.audio</li>
<li>octavePackages.bim</li>
<li>octavePackages.bsltl</li>
<li>octavePackages.cgi</li>
<li>octavePackages.communications</li>
<li>octavePackages.control</li>
<li>octavePackages.data-smoothing</li>
<li>octavePackages.database</li>
<li>octavePackages.dataframe</li>
<li>octavePackages.dicom</li>
<li>octavePackages.divand</li>
<li>octavePackages.doctest</li>
<li>octavePackages.econometrics</li>
<li>octavePackages.financial</li>
<li>octavePackages.fits</li>
<li>octavePackages.fpl</li>
<li>octavePackages.fuzzy-logic-toolkit</li>
<li>octavePackages.ga</li>
<li>octavePackages.general</li>
<li>octavePackages.generate_html</li>
<li>octavePackages.geometry</li>
<li>octavePackages.gsl</li>
<li>octavePackages.image</li>
<li>octavePackages.image-acquisition</li>
<li>octavePackages.instrument-control</li>
<li>octavePackages.interval</li>
<li>octavePackages.io</li>
<li>octavePackages.level-set</li>
<li>octavePackages.linear-algebra</li>
<li>octavePackages.lssa</li>
<li>octavePackages.ltfat</li>
<li>octavePackages.mapping</li>
<li>octavePackages.matgeom</li>
<li>octavePackages.miscellaneous</li>
<li>octavePackages.msh</li>
<li>octavePackages.mvn</li>
<li>octavePackages.nan</li>
<li>octavePackages.ncarray</li>
<li>octavePackages.netcdf</li>
<li>octavePackages.nurbs</li>
<li>octavePackages.ocl</li>
<li>octavePackages.octclip</li>
<li>octavePackages.octproj</li>
<li>octavePackages.optics</li>
<li>octavePackages.optim</li>
<li>octavePackages.optiminterp</li>
<li>octavePackages.parallel</li>
<li>octavePackages.quaternion</li>
<li>octavePackages.queueing</li>
<li>octavePackages.signal</li>
<li>octavePackages.sockets</li>
<li>octavePackages.sparsersb</li>
<li>octavePackages.splines</li>
<li>octavePackages.statistics</li>
<li>octavePackages.stk</li>
<li>octavePackages.strings</li>
<li>octavePackages.struct</li>
<li>octavePackages.symbolic</li>
<li>octavePackages.tisean</li>
<li>octavePackages.tsa</li>
<li>octavePackages.video</li>
<li>octavePackages.windows</li>
<li>octavePackages.zeromq</li>
<li>octopus</li>
<li>openems</li>
<li>python38Packages.python-csxcad</li>
<li>python38Packages.python-openems</li>
<li>python39Packages.python-csxcad</li>
<li>python39Packages.python-openems</li>
</ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/development/libraries/science/math/arpack/default.nix [hunk 2]
```diff
@@ -24,8 +34,7 @@ stdenv.mkDerivation rec {
 
   cmakeFlags = [
     "-DBUILD_SHARED_LIBS=ON"
-    "-DINTERFACE64=${lib.optionalString blas.isILP64 "1"}"
-  ];
+  ] ++ lib.optional blas.isILP64 "-DINTERFACE64=1";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/science/math/arpack/default.nix:37]:
```suggestion
  ] ++ lib.optionals blas.isILP64 [ "-DINTERFACE64=1" ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/science/math/arpack/default.nix (hunk 2 above)
Comment (doronbehar) [reply pkgs/development/libraries/science/math/arpack/default.nix:37]: Speaking of which, I'd like:

```
    "-DINTERFACE64=${if blas.isIL64 then 1 else 0}"
```

**Review by doronbehar — APPROVED** [review]

**Comment by doronbehar** [comment]
Thanks for joining maintenance on this :)

**Pull request merged**